Fix problem with sparse layers in tf0.1.0

cryu854 opened the pull request:
The `_resource_apply_sparse` function applies update according to the indices. Gathering the elements before update would fix the error caused by ResourceScatterAdd. The fix has been tested on [word embeddings](https://www.tensorflow.org/tutorials/text/word_embeddings).

cryu854 committed:
Make it compatible with tensroflow 2.3

cryu854 committed:
modified from tensorflow/tensorflow/python/keras/optimizer_v2/adam.py

andreselizondo-adestech committed:
Adds TF-Keras compatibility to pypi package.

cryu854 committed:
Merge pull request #1 from andreselizondo-adestech/master

Adds TF-Keras compatibility to pypi package.

cryu854 committed:
Add eps to bias correction & modify apply_dense

cryu854 committed:
Fix bug with sparse layers

cryu854 committed:
Fix tf.gather NameError

cryu854 committed:
Merge branch 'update_0.1.0' of https://github.com/juntang-zhuang/Adabelief-Optimizer into update_0.1.0

cryu854 committed:
Fix problem with sparse layers in tf

cryu854 committed:
Merge tf0.0.9 with origin

cryu854 committed:
Update __init__.py

cryu854 committed:
Update setup.py

juntang-zhuang commented:
@cryu854 Hi, just curious to check if you are still interested in contributing to the tensorflow-addons project? I think you are much better than me for this task because you have much more experience with Tensorlfow. Please see a feature request under tensorflow-addons project: https://github.com/tensorflow/addons/issues/2203, if you are interested we can create a pull request and upload Adabelief code based on current version.

cryu854 commented:
@juntang-zhuang Yes, I would like to help with this. Here is a tensorflow-addons's [CONTRIBUTING.md](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md) doc, explaining the review process and how to write tests for the contribution. In order to run the tests, we may also need to set up the development environment too. I'll look into the details of some tensorflow-addons projects and let you know when I have any progress.

juntang-zhuang commented:
Thanks so much for help!

cryu854 commented:
@juntang-zhuang Hi, I've implemented the test code for AdaBelief, the code `adabelief_test.py` is a combination of `lamb_test.py` and `rectified_adam_test.py`. Please take a look at [my fork of addons](https://github.com/cryu854/addons), I will take any suggestion or modification if anything else is needed. And if all goes well, we can make a PR to addons to conduct further code reviews.

juntang-zhuang commented:
@cryu854 Nice work! Thanks so much for help. I quickly tested it and it works quite good.

cryu854 commented:
@juntang-zhuang I've made this PR [#2234](https://github.com/tensorflow/addons/pull/2234#issue-517509036) to addons. Thank you for giving me this opportunity to contribute to tensorflow-addons.

cryu854 committed:
Remove tf.control_dependencies

cryu854 committed:
Add unit test for AdaBelief_tf

cryu854 committed:
Modify AdaBelief to AdaBeliefOptimizer

cryu854 committed:
Remove addons dependencies

cryu854 committed:
Merge branch 'update_0.1.0' into tf0.1.0

cryu854 committed:
Update README.md

cryu854 commented:
@juntang-zhuang Hi, I think the PR for addons will be postponed for a while, so I did the following modifications for current tensorflow version:
- Since tensorflow 2 uses Eager mode and AutoGraph, there is no need to ensure the assignment dependencies within the context manager `tf.control_dependencies`.
- Removed register and addons dependencies.
- Added unit tests for `AdaBelief.py`.
- Fixed some typos at `README.md` and `setup.py`.

Hope these help!

cryu854 committed:
Add example of serialization

juntang-zhuang commented:
@cryu854 Thanks so much, I just merged your pull request. Thanks gain for your contribution.
